src/legacy/core_plugins/vega/public/data_model/es_query_parser.js

```diff
@@ -146,13 +147,27 @@ export class EsQueryParser {
     return { dataObject, url };
   }
 
+  mapRequest = (request) => {
+    const esRequest = request.url;
+    if (this._esShardTimeout) {
+      const { body = {}, ...rest } = esRequest;
+      body.timeout = `${this._esShardTimeout}ms`;
```

Comment:
I think it should check if the user has defined a custom timeout as part of the vega, e.g.
```
url: {
  index: logstash
  body: {...}
  timeout: 10s
}
```

Reply:
I don't think they should be able to do that, why I think we should leave it that way (or actually make sure again, they can't overwrite it). The `elasticsearch.shardTimeout` is a `kibana.yml` setting for a reason: because we want administrators to decide about it and not users, since it can effect the full cluster performance. So we should make sure that the users can't mess with that setting. But we def need to test that a user can't overwrite that timeout anymore if one was set in the `kibana.yml`.

Reply:
@timroes should we allow user to make it "shorter than the main timeout" ?  Since this is a query written directly by the user, they may want to shorten it to prevent too long of a query...?  Also, I think in the "trusted user scenario", the main timeout might be actually less than what a user may want for a specific query - e.g. in general case, timeout would be 1min, but for some complex graph, user might be ok to run it for 10 minutes to generate a complex report. Should we have a "allow Vega user custom timeout" setting in the `kibana.yml`?  Perhaps this might be a separate PR.

Reply:
@nyurik @timroes I moved this discussion into https://github.com/elastic/kibana/issues/30623, made sure there is no user provided timeout in the request (only `kibana.yml` setting is repected) and added a test for that. Should we add a note to the docs about that?

Reply:
I don't think we need to have that note for now and let's move discussion over to the newly created issue.